numba_dpex/core/datamodel/models.py

```diff
@@ -40,18 +40,72 @@ def __init__(self, dmm, fe_type):
             ),
             (
                 "parent",
-                types.CPointer(fe_type.dtype, addrspace=fe_type.addrspace),
+                types.CPointer(types.pyobject, addrspace=fe_type.addrspace),
             ),
             ("nitems", types.intp),
             ("itemsize", types.intp),
             (
                 "data",
                 types.CPointer(fe_type.dtype, addrspace=fe_type.addrspace),
             ),
+            (
+                "sycl_queue",
+                types.CPointer(types.void, addrspace=fe_type.addrspace),
+            ),
+            ("shape", types.UniTuple(types.intp, ndim)),
+            ("strides", types.UniTuple(types.intp, ndim)),
+        ]
+        super(USMArrayModel, self).__init__(dmm, fe_type, members)
+
+
+class DpnpNdArrayModel(StructModel):
+    """Data model for the DpnpNdArray type.
+
+    The data model for DpnpNdArray is similar to numb's ArrayModel used for
+    the numba.types.Array type, with the additional field ``sycl_queue`. The
```

Comment:
>The `sycl_queue` attribute stores the pointer to the C++ sycl::queue object that was used to allocate memory for numba-dpex's native representation for an Python object inferred as a DpnpNdArray.

The `sycl_queue` attribute stores the pointer to the C++ sycl::queue object that allocates memory for numba-dpex's native representation of a Python object inferred as a DpnpNdArray.

or a better, clearer explanation.

Reply:
How about: 

>
    DpnpNdArrayModel is used by the numba_dpex.types.DpnpNdArray type and
    abstracts the usmarystruct_t C type defined in
    numba_dpex.core.runtime._usmarraystruct.h.

    The DpnpNdArrayModel differs from numba's ArrayModel by including an extra
    member sycl_queue that maps to _usmarraystruct.sycl_queue pointer. The
    _usmarraystruct.sycl_queue pointer stores the C++ sycl::queue pointer that
    was used to allocate the data for the dpnp.ndarray represented by an
    instance of _usmarraystruct.